WordPress/Classes/ViewRelated/Blog/Site Picker/AddSiteController.swift

```diff
@@ -15,6 +15,12 @@ struct AddSiteController {
     }
 
     func showDotComSiteCreationScreen() {
+        if let account = try? WPAccount.lookupDefaultWordPressComAccount(in: ContextManager.shared.mainContext),
+           account.needsEmailVerification {
+            VerifyEmailModal.present(on: viewController)
+            return
+        }
```

Comment:
I think there is an issue here. The `needsEmailVerification` will be checked just locally from Core Data. So, if I verify the email and keep the app open, but also if I dismiss the verify email modal and reopen it, we do not check again if the email has been verified, so I'm stuck there. Therefore, I would suggest verifying every time the user taps to create a new site if the email needs verification. Additionally, add a button in `VerifyEmailView` that allows the user to confirm they verified their email. This way, we can check remotely again if the email has been verified, and this view can be dismissed.